lib/internal/crypto/cipher.js

```diff
@@ -177,6 +199,8 @@ Cipher.prototype.update = function update(data, inputEncoding, outputEncoding) {
       'data', ['string', 'Buffer', 'TypedArray', 'DataView'], data);
   }
 
+  validateInputEncoding(inputEncoding);
```

Comment:
> `If data is a [Buffer](https://nodejs.org/api/buffer.html), TypedArray, or DataView, then inputEncoding is ignored.`

This needs to go to the `if (typeof data === 'string') {` condition where we already validate encoding. So I wonder, given we have other uses of validateEncoding that from the looks of it actually validates data length if encoding is `hex`, do we have a larger problem here?

cc @nodejs/crypto

Reply:
There's a lot of undocumented and legacy behaviour to navigate around here.

> If the inputEncoding argument is not given, data must be a [Buffer](https://nodejs.org/api/buffer.html), TypedArray, or DataView.

And yet there are tests dating back to 0.x (https://github.com/nodejs/node-v0.x-archive/issues/5482, f59ab10a6415e7dce7a6c3b6e734c213298ec205) that check for this.